sapp/ui/frontend/src/Filter.js

```diff
@@ -187,6 +187,86 @@ const Paths = (
   );
 };
 
+const MatchesIsField = (
+  props: $ReadOnly<{
+    allOptions: $ReadOnlyList<string>,
+    parameterName: string,
+    currentFilter: FilterDescription,
+    setCurrentFilter: FilterDescription => void,
+  }>,
+): React$Node => {
+  const [inputValue, setInputValue] = useState('');
+  const [selectValue, setSelectValue] = useState([]);
+  const [matchedValues, setMatchedValues] = useState([]);
+
+  const triggerChange = (operation: string, value: mixed = undefined): void => {
+    let values = {};
+    if(value === undefined) {
+      value = (operation === 'is')? selectValue : inputValue;
+    }
+    values[props.parameterName] = {
+      operation: operation,
+      value: value,
+    };
+    props.setCurrentFilter({...props.currentFilter, ...values});
+  }
+
+  const selectedOperation =
+    (props.currentFilter[props.parameterName]?.operation !== undefined)?
+      props.currentFilter[props.parameterName].operation : 'is';
+
+  return (
+    <Row gutter={gutter}>
+      <Col span={6}>
+        <Select
+          options={[{value: 'is'}, {value: 'matches'}]}
+          value={selectedOperation}
+          onChange={triggerChange}
+          style={{width: '100%'}}
+        />
+      </Col>
+      <Col span={16}>
+        {selectedOperation === 'matches' ? (
+          <Input
+            placeholder="regular expression"
+            style={{width: '100%'}}
+            value={inputValue}
+            onChange={event => {
+              const value = event.target.value;
+              setInputValue(value);
+              const option_values = props.allOptions.filter(option =>
+                  option.match(value),
```

Comment:
Do Python and JavaScript use the same Regex engines? If not, trying to run the same regex on both might end up with confusing subtleties where frontend results don't match backend results

Reply:
I never thought we would have large differences, but it seems the case! So I updated the backend to use a library called js_regex!